pytest_plugins/metadata_markers.py

```diff
@@ -100,6 +111,16 @@ def pytest_collection_modifyitems(session, items, config):
             ("start_time", datetime.datetime.utcnow().strftime(FMT_XUNIT_TIME))
         )
 
+        # add custom ibutsu metadata fields for test case grouping and heatmaps
+        if hasattr(item, "_ibutsu"):
+            item._ibutsu["data"]["metadata"].update(
+                {
+                    # TODO Work with ibutsu team, better mechanism for defining 'special' data
+                    # TODO Add sat version to this item data at test execution time
+                    "component": item.get_closest_marker('component').args[0]
```

Comment:
will this help in search the components in ibutsu ? Along with this can we also include the username (as we have that in testimony token as well )

Reply:
user_properties are already stored on the test results in ibutsu under metadata, allowing for filtering by assignee (or any of the other values that are added to user_properties through this hook already.

Ibutsu treats `component` and `env` as special metadata values, and requires them to be set for Jenkins analysis and heatmap widgets to render.  This just sets the field explicitly so these widgets will start to work for us.

I'm working with the ibutsu team to also re-work the widget dependency on these fields, and to add more filtering capability to the UI.